account_reconcile_blue_lines_restriction/models/account_reconcile_model.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import models
+
+
+class AccountReconcileModel(models.Model):
+
+    _inherit = "account.reconcile.model"
+
+    def _apply_conditions(self, query, params):
+        """Remove AMLs with account not allowed for reconciliation from
+           blue lines appearance conditions"""
+        new_query, new_params = super()._apply_conditions(query, params)
+        if self.rule_type == "invoice_matching":
+            new_query += """
+                AND (
+                    -- modified blue lines appearance conditions
+                    aml.account_id IN (
+                        journal.default_credit_account_id,
+                        journal.default_debit_account_id
+                    )
+                    AND aml.statement_id IS NULL
+                    AND (
+                        company.account_bank_reconciliation_start IS NULL
+                        OR
+                        aml.date > company.account_bank_reconciliation_start
+                    )
+                    -- This line restricts propositions to lines on
+                    -- accounts allowing reconciliation
+                    AND account.reconcile IS TRUE
+                    OR (
+                        -- black lines appearance conditions
+                        account.reconcile IS TRUE
+                        AND aml.reconciled IS FALSE
```

Comment:
this condition should already be in the query? I do not really understand how it should work

Reply:
Yes this part is already there, however it's inside this big `AND` condition: https://github.com/odoo/odoo/blob/13.0/addons/account/models/account_reconcile_model.py#L575-L592
Therefore we must redefine here this whole condition to add `AND account.reconcile IS TRUE`.

I must still add tests though.